tests/src/PatternAssertion/FilePatternAssert.php

```diff
@@ -0,0 +1,142 @@
+   *   The crawler.
+   */
+  protected function assertFile(array $expected, Crawler $crawler): void {
+    $file = $crawler->filter('body > div.mt-4 > div.border.rounded > div:first-child');
+    self::assertCount(1, $file);
+
+    (new IconPatternAssert())->assertPattern([
+      'name' => $expected['icon'],
+      'size' => '2xl',
+    ], $crawler->filter('svg.icon--file')->outerHtml());
+    $this->assertItem($expected, $file);
+  }
+
+  /**
+   * Asserts the translations section of the pattern.
+   *
+   * @param array|null $expected
+   *   The expected translation section values, or NULL if no translations are
+   *   expected.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The crawler.
+   */
+  protected function assertTranslations(?array $expected, Crawler $crawler): void {
+    // This selector targets the DIVs that wrap the main file and the
+    // translations. When no translations are present, there should be only one
+    // node.
+    $container_selector = 'body > div.mt-4 > div.border.rounded > div';
+
+    if ($expected === NULL) {
+      // When no translations are present, only one div exists inside the
+      // container.
+      $this->assertElementExists($container_selector, $crawler);
+      $this->assertElementNotExists('.collapse', $crawler);
+      return;
+    }
+
+    $containers = $crawler->filter($container_selector);
+    self::assertCount(2, $containers);
+    // The second node wraps all the translations section.
+    $wrapper = $containers->eq(1);
+    $this->assertElementText(sprintf('Other languages (%s)', count($expected)), '.text-end a', $wrapper);
+
+    $translation_nodes = $wrapper->filter('.collapse > div');
+    self::assertSameSize($expected, $translation_nodes);
+    foreach ($expected as $index => $translation) {
+      $this->assertItem($translation, $translation_nodes->eq($index));
+    }
+  }
+
+  /**
+   * Asserts the label used for the download links.
+   *
+   * @param string $expected
+   *   The expected label.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The crawler.
+   */
+  protected function assertLinksLabel(string $expected, Crawler $crawler): void {
+    $this->assertElementText($expected, 'body > div.mt-4 > div.border.rounded > div:first-child a', $crawler);
+    foreach ($crawler->filter('.collapse > div a') as $node) {
+      self::assertEquals($expected, $node->textContent);
+    }
+  }
+
+  /**
+   * Asserts elements that are common for file and translation sections.
+   *
+   * @param array $expected
+   *   The expected values.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The crawler.
+   */
+  protected function assertItem(array $expected, Crawler $crawler): void {
+    $this->assertElementExists(sprintf('a[href="%s"]', $expected['url']), $crawler);
+    $this->assertElementText($expected['title'], 'p.fw-bold.m-0', $crawler);
+
+    $data_container = $crawler->filter('p.fw-bold.m-0 + small.fw-bold');
```

Comment:
dono how this passed here, but in whitelabel when we use it, as soon as we add translations this fails. This is because `p.fw-bold.m-0 + small.fw-bold` fetches the first small after the mentioned p, but the translations have a `<p class="m-0"></p>` in between so $data_container is null.
```suggestion
    $data_container = $crawler->filter('small.fw-bold');
```

Reply:
Wrote it as comment in the ticket: on whitelabel the patch is not applied because we are not using a release, so there it's mean to fail until we make a temporary release.